File: Matlab/PlotWaveFunction.m
%Script for plotting Rydberg radial wavefunctions
%24/07/2017

close all
clear
clc

%Input information
atom = '87Rb';
nn = 5;
ll = 0;
jj = 0.5;

%Get some constants
SIunits;
if strcmp(atom,'87Rb')==1
    Rb87numbers;
end

%Calculate wave function
[normY_sol,rr] = numerovfunc(atom,nn,ll,jj);

%Rescale for plotting
plotscale = sqrt(rr);
probamp = (normY_sol.*plotscale).^2;

figure;
plot(plotscale,normY_sol,'LineWidth',2)
if nn > 20
    xlim([sqrt(alpha_c^(1/3)),max(plotscale)]);
end
xlabel('$(r/a_0)^{1/2}$','interpreter','latex');
ylabel('$r^{1/2}R(r)\, (a_0^{-1})$','interpreter','latex');
title([atom,' radial wavefunction |n,l,j\rangle = |',num2str(nn),',',...
    num2str(ll),',',num2str(jj),'\rangle']);
set(gcf,'Color','w');

figure;
plot(rr,probamp,'LineWidth',2)
if nn > 20
    xlim([alpha_c^(1/3),max(rr)]);
end
xlabel('$r/a_0$','interpreter','latex');
ylabel('$|r^{1/2}R(r)|.^2\, (a_0^{-1})$','interpreter','latex');
title([atom,' radial wavefunction |n,l,j\rangle = |',num2str(nn),',',...
    num2str(ll),',',num2str(jj),'\rangle']);
set(gcf,'Color','w');

File: Julia/PlotWaveFunction.jl
#Script to plot Rydberg radial wave functions
#23/07/2017

using Plots, JLD, LaTeXStrings
pyplot()
include("functions.jl")
PyPlot.close("all")

#Input information
atom = "87Rb"
nn = 50
ll = 0
jj = 0.5

#Calculate wave function
normY_sol, rr = numerovfunc(atom,nn,ll,jj)

#Rescale for plotting
plotscale = sqrt(rr)
probamp = (normY_sol.*plotscale).^2

alpha_c = getalpha(atom)

PyPlot.figure()
if nn>20
  plot(plotscale[plotscale .> sqrt(alpha_c^(1/3))],normY_sol[plotscale .> sqrt(alpha_c^(1/3))],linewidth=2)
else
  plot(plotscale,normY_sol,linewidth=2)
end
plot!(xlabel=L"(r/a_0)^{1/2}")
plot!(ylabel=L"r^{1/2}R(r) \,(a_0^{-1})")
plot!(title=string(atom," radial wavefunction |n,l,j⟩ = |",string(nn),",",string(ll),",",string(jj),"⟩"))
plot!(leg=false)
gui()

PyPlot.figure()
if nn>20
  plot(rr[rr .> alpha_c^(1/3)],probamp[rr .> alpha_c^(1/3)],linewidth=2)
else
  plot(rr,probamp,linewidth=2)
end
plot!(xlabel=L"r/a_0")
plot!(ylabel=L"|rR(r)|^2 \,(a_0^{-1})")
plot!(title=string(atom," radial probability density |n,l,j⟩ = |",string(nn),",",string(ll),",",string(jj),"⟩"))
plot!(leg=false)
gui()
